third_party/user/user.go

```diff
@@ -378,7 +378,7 @@ func GetExecUser(userSpec string, defaults *ExecUser, passwd, group io.Reader) (
 
 				// Okay, so it's numeric. We can just roll with this.
 			}
-		} else if len(groups) > 0 {
+		} else if len(groups) > 0 && uidErr != nil {
```

Comment:
While I will merge this, note that this code comes from `runc` and I really don't want us to diverge from upstream (the reason I didn't just vendor it is because I needed to patch parts of it IIRC).

Can you elaborate why this change is necessary? Is it to avoid the supplementary groups stuff if the user is numeric? Maybe we should have this same patch put into runc?